pixels/canvas.py

```diff
@@ -55,7 +55,7 @@ async def _populate_cache(self, conn: Connection) -> None:
     async def is_cache_out_of_date(self, conn: Connection) -> bool:
         """Return true if the cache can be considered out of date."""
         cache = await self.get_pixels()
-        if len(cache)//3 != constants.width*constants.height:
+        if cache and len(cache)//3 != constants.width*constants.height:
```

Comment:
```suggestion
        if not cache or len(cache)//3 != constants.width*constants.height:
```
I think this short circuit will do what we want